test/runtests.jl

```diff
@@ -261,4 +253,70 @@ end
 
       @test cal_energy(sol)/(x[1]-x0[1]+x[2]-x0[2]) ≈ 1e5
    end
+
+   @testset "normalized field" begin
+      # 3D
+      x = range(-10, 10, length=15)
+      y = range(-10, 10, length=20)
+      z = range(-10, 10, length=25)
+      B = fill(0.0, 3, length(x), length(y), length(z)) # [B₀]
+      E = fill(0.0, 3, length(x), length(y), length(z)) # [v₀B₀]
+
+      B₀ = 10e-9
+
+      B[3,:,:,:] .= 1.0
+
+      Δx = x[2] - x[1]
+      Δy = y[2] - y[1]
+      Δz = z[2] - z[1]
+
+      mesh = CartesianGrid((length(x)-1, length(y)-1, length(z)-1),
+         (x[1], y[1], z[1]),
+         (Δx, Δy, Δz))
+
+      x0 = [0.0, 0.0, 0.0] # initial position, [m]
+      u0 = [1.0, 0.0, 0.0] # initial velocity, [m/s]
+      stateinit = [x0..., u0...]
+
+      param = prepare(mesh, E, B, B₀; species=Proton)
+      tspan = (0.0, 1.0)
+
+      prob = ODEProblem(trace_normalized!, stateinit, tspan, param)
+
+      sol = solve(prob, Tsit5(); save_idxs=[1])
+
+      x = getindex.(sol.u, 1)
+
+      @test length(x) == 8 && x[end] ≈ 0.8540967226941674
+
+      # 2D
+      x = range(-10, 10, length=15)
+      y = range(-10, 10, length=20)
+      B = fill(0.0, 3, length(x), length(y)) # [B₀]
+      E = fill(0.0, 3, length(x), length(y)) # [v₀B₀]
+
+      B₀ = 10e-9
+
+      B[3,:,:,:] .= 1.0
```

Comment:
Because of the decrease of the dimensions, this line should probably be rewritten as
```julia
B[3,:,:] .= 1.0
```

Reply:
You are right. It a surprise that Julia does not complain about

```julia
a = zeros(3,2,2)
a[3,:,:,:,:,:] # as many singular dimension as you want
```

Is this a hidden feature?

Reply:
I am surprised too. But the types of `B[3,:,:]` and `B[3,:,:,:]` are different, maybe it is a hidden feature.